include/swift/AST/Types.h

```diff
@@ -4401,6 +4401,64 @@ class SILTokenType final : public TypeBase {
 };
 DEFINE_EMPTY_CAN_TYPE_WRAPPER(SILTokenType, Type)
 
+// SWIFT_ENABLE_TENSORFLOW
+class SILDifferentiableFunctionType;
+typedef CanTypeWrapper<SILDifferentiableFunctionType>
+    CanSILDifferentiableFunctionType;
+
+/// The SIL-only type for differentiable functions, which represent a bundle of
+/// the original function and autodiff-asssociated functions. A SIL
+/// differentiable function type stores 3 types: the original function type, the
+/// type of the first-order differential and the type of the first-order
+/// pullback. These types help keep track of the abstraction pattern of all
+/// autodiff-associated functions. The generic signature of this function
+/// redirects to that of the original function, which serves as the source of
+/// truth.
+class SILDifferentiableFunctionType final
+    : public TypeBase, public llvm::FoldingSetNode {
+  int maxOrder;
```

Comment:
I think it might be useful (or even necessary) to have the wrt indices in here too.

Reply:
Good point. This can help remove the `@nondiff` attribute.